pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -136,6 +139,11 @@ func (p *Provider) Init() error {
 		return fmt.Errorf("error while parsing default rule: %w", err)
 	}
 
+	// Set the default health check to keep container to "passing"
+	if len(p.StrictChecks) == 0 {
+		p.StrictChecks = []string{api.HealthPassing, api.HealthWarning}
```

Comment:
You got it right in `SetDefaults`, but I don't think we should set the default values here. 
If `StrictChecks` is empty at this stage, it's deliberately the choice of the user. I don't see why someone would want to do it, but from a user perspective, the behavior is easier to understand.

Reply:
I wasn't seeing `SetDefaults` called anywhere, and noticed that tests were failing, which told me the defaults weren't actually being set, so, I added it here. Are we missing a call to `SetDefaults` somewhere?

Reply:
Oh i see, defaults arent set during tests, but are set in the main flow, so just need to make sure we include that in the tests.

Reply:
Yes indeed, `SetDefaults` is not explicitly called in the code. When building the static configuration we recursively call this method via reflect. For your test, it would be best to call this method manually on the test config.